language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/ContractIdClass.scala

```diff
@@ -0,0 +1,254 @@
+package com.daml.lf.codegen.backend.java.inner
+
+import com.daml.ledger.javaapi
+import com.daml.lf.codegen.backend.java.ObjectMethods
+import com.daml.lf.data.Ref.PackageId
+import com.daml.lf.iface.Type
+import com.squareup.javapoet._
+import scala.jdk.CollectionConverters._
+
+import java.util.Optional
+import javax.lang.model.element.Modifier
+
+object ContractIdClass {
+
+  def builder(
```

Comment:
Do you have tests for this builder?

Reply:
Funnily I can now say _yes_! I just renamed the `TemplateClassSpec` to `ContractIdClassBuilderSpec` (the names get horrible long xD), because the tests in there turned out to be only related to code which was moved out into the `ContractIdClass.scala`.

Reply:
Great!